Userland/Libraries/LibWeb/IndexedDB/Internal/ConnectionQueueHandler.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024, stelar7 <dudedbz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <LibJS/Heap/Handle.h>
+#include <LibWeb/DOM/EventTarget.h>
+#include <LibWeb/IndexedDB/IDBRequest.h>
+#include <LibWeb/StorageAPI/StorageKey.h>
+
+namespace Web::IndexedDB {
+
+using ConnectionQueue = AK::Vector<JS::Handle<IDBRequest>>;
+using ConnectionMap = HashMap<StorageAPI::StorageKey, HashMap<String, ConnectionQueue>>;
+
+// https://w3c.github.io/IndexedDB/#connection-queues
+class ConnectionQueueHandler {
+public:
+    static ConnectionQueue& for_key_and_name(StorageAPI::StorageKey& key, String& name);
+    static ConnectionQueueHandler the()
```

Comment:
Is this intended to return a new copy each time?

Reply:
Yes and no... The instance is static, so its the same every time?
(If its not a copy, some WPT tests start to timeout 🤔 )